File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) 
    {
        int size=nums.size();
        int rot=k%size;
        vector<int>ans;
        for(int i=0;i<rot;i++)
        {
            ans.push_back(nums[size-rot+i]);
        }
        for(int i=0;i<size-rot;i++)
        {
            ans.push_back(nums[i]);
        }
        nums=ans;
    }
};

File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) 
    {
       int size=nums.length;
       int rot=k%size;
       int[] ans=new int[size];
       for(int i=0;i<rot;i++)
       {
          ans[i]=nums[size-rot+i];
       }
       for(int i=0;i<size-rot;i++)
       {
           ans[i+rot]=nums[i];
       }
       for(int i=0;i<size;i++)
       {
           nums[i]=ans[i];
       }
    }
}